### DISCUSSION TITLE
isWithinInterval exclution criteria

### DISCUSSION THREAD

**Discussion by p-prins** [js-fns/date-fns#2818] [v3 API ideas]
Part of the definition of an interval is whether or not the start and the end should be included or not.

[ start, end ] (start and end included)
[ start, end > (start included, end excluded)
< start, end > (start and end excluded)

Currently the isWithinInterval function only allows us to specify the type of interval where the start and end is included. I propose to support the other two types of intervals as well. 

To accomplish this I'd like to add a third parameter to the function named options (similar to for example eachDayOfInterval). Options accepts an object which would accept two properties `excludeStart` and `excludeEnd`.

```
type options = {
  excludeStart: boolean;
  excludeEnd: boolean;
}
``` 

I'm happy to make a PR if this will be accepted. Of course, I'm open to other ideas as well on how to implement the proposed feature.

**Comment by tan75** [comment]
hi @p-prins 
I will convert this to a discussion and bring this up at our maintainer meeting.

**Comment by tan75** [comment]
hi @p-prins
this is a very good idea however currently we are focused on migration to TS and v3 release. 
Fot this reaseon we marked this as a v3 API idea. 
Could you please hold off for a while and wait for the v3 release? Once the v3 is out, you can start working on the PR if you please.
